java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -416,56 +447,19 @@ private boolean isArgumentMatchesFlagName(
    * 4. Qualified enum name with method call (e.g. "TestExperimentName.STALE_FLAG.getKey()")
    */
-  private boolean isMatchingEnumFieldValue(ExpressionTree argTree, VisitorState state) {
-    // Skip if no enum records are configured, to avoid excessive AST lookups
-    if (!this.config.hasEnumRecords()) {
-      return false;
-    }
-
-    Symbol sym;
-    switch (argTree.getKind()) {
-      case IDENTIFIER:
-        // Case 1: Unqualified enum constant (e.g. "STALE_FLAG")
-        sym = ASTHelpers.getSymbol(argTree);
-        break;
-      case MEMBER_SELECT:
-        // Case 2: Qualified enum name (e.g. "TestExperimentName.STALE_FLAG")
-        MemberSelectTree memberTree = (MemberSelectTree) argTree;
-        sym = ASTHelpers.getSymbol(memberTree);
-        break;
-      case METHOD_INVOCATION:
-        // Because the enum constant itself is being removed, we can remove any method called from
-        // that enum constant and not only specific, configured ones
-        MethodInvocationTree methodTree = (MethodInvocationTree) argTree;
-        if (methodTree.getMethodSelect().getKind() != Kind.MEMBER_SELECT) {
-          return false;
-        }
-        MemberSelectTree methodMemberTree = (MemberSelectTree) methodTree.getMethodSelect();
-        if (methodMemberTree.getExpression().getKind() == Kind.IDENTIFIER) {
-          // Case 3: Unqualified enum constant with method call (e.g. "STALE_FLAG.getKey()")
-          IdentifierTree identifier = (IdentifierTree) methodMemberTree.getExpression();
-          sym = ASTHelpers.getSymbol(identifier);
-        } else if (methodMemberTree.getExpression().getKind() == Kind.MEMBER_SELECT) {
-          // Case 4: Qualified enum name with method call (e.g.
-          // "TestExperimentName.STALE_FLAG.getKey()")
-          MemberSelectTree memberTreeExpression =
-              (MemberSelectTree) methodMemberTree.getExpression();
-          sym = ASTHelpers.getSymbol(memberTreeExpression);
-        } else {
-          return false;
-        }
-        break;
-      default:
-        return false;
-    }
-
-    if (sym.getKind() != ElementKind.ENUM_CONSTANT) {
-      return false;
-    }
-
-    Symbol.ClassSymbol enclosingClass = ASTHelpers.enclosingClass(sym);
-
-    return isEnumConstantMatchingFlagName(sym.getSimpleName().toString(), enclosingClass, state);
+  private boolean isMatchingEnumFieldValue(ExpressionTree tree, VisitorState state) {
+    // Handles - Unqualified import enum constant
+    Matcher<ExpressionTree> enumFieldValueMatcher =
+        symbolMatcher(
+            (sym, st) ->
+                isEnumConstantMatchingFlagName(
+                    sym.getSimpleName().toString(), ASTHelpers.enclosingClass(sym), state));
+    // Handles - the other three scenarios
+    return anyOf(
+            enumFieldValueMatcher,
+            Matchers.methodInvocation(
+                anyOf(enumFieldValueMatcher, memberSelectExpression(enumFieldValueMatcher))))
```

Comment:
@lazaroclapp  I addressed your comments regarding the new check being too aggressive. Now it is equivalent to the switch case check from before.